msgcntr/messageforums-app/src/java/org/sakaiproject/tool/messageforums/ui/MessageForumStatisticsBean.java

```diff
@@ -1849,43 +1998,61 @@ public int compare(Object item, Object anotherItem){
 			public int compare(Object item, Object anotherItem){
 				DecoratedCompiledMessageStatistics stat = ((DecoratedCompiledMessageStatistics) item);
 				DecoratedCompiledMessageStatistics stat2 = ((DecoratedCompiledMessageStatistics) anotherItem);
-				
-				if(stat.getGradebookAssignment() == null || !stat.getGradebookAssignment().isAllowedToGrade())
-					return 1;
-				
-				if(stat2.getGradebookAssignment() == null || !stat2.getGradebookAssignment().isAllowedToGrade())
-					return -1;
-				
-				if(stat.getGradebookAssignment().getScore() == null || "".equals(stat.getGradebookAssignment().getScore()))
-					return 1;
-				
-				if(stat2.getGradebookAssignment().getScore() == null || "".equals(stat.getGradebookAssignment().getScore()))
-					return -1;
-				
-				if(!isNumber(stat.getGradebookAssignment().getScore()))
-					return 1;
-				
-				if(!isNumber(stat2.getGradebookAssignment().getScore()))
-					return -1;
-				
-				try{
-					double val = Double.valueOf(stat.getGradebookAssignment().getScore()).doubleValue() - Double.valueOf(stat2.getGradebookAssignment().getScore()).doubleValue();
-					if(val < 0){
-						return 1;
-					}else{
-						return -1;
-					}
-				}catch(NumberFormatException e){
-				}
-				
-				//we can't have descrepancies on how the order happens, otherwise jsf will scramble the scores
-				//with other scores that are equal to this (jsp submits twice, causing "sort" to happen between when
-				//the user enters the data and when it gets submitted in JSP (behind the scenes)
-				return nameComparatorAsc.compare(item, anotherItem);
-				
+
+				return 0 - compareGradesFromStats(stat, stat2);
 			}
 		};
 	}
+
+	private static int compareGradesFromStats(DecoratedCompiledMessageStatistics stat1, DecoratedCompiledMessageStatistics stat2)
```

Comment:
Nice work extracting the above duplicated code into methods. Some JavaDocs would be nice though.